MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,787 @@
+# Migration Guide: React Native New Architecture
```

Comment:
Is this effectively the README or is there also a README somewhere? Can we rename this to README?